java.lang.NoClassDefFoundError after upgrading mysql to 1.11.1

I am creating a `MySQLContainer` with an init script
```
public static MySQLContainer mysql = (MySQLContainer) new MySQLContainer().withInitScript("sql/create_schema.sql");
```

This results in the following exception
```
Caused by: java.lang.ClassNotFoundException: org.apache.commons.lang.StringUtils
	at java.net.URLClassLoader.findClass(URLClassLoader.java:381)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
	at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:349)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
	... 24 more
```

This instantiation worked fine on 1.10.7

I checked with Slack and they suggested I report this as the shading changing in 1.11

--- a/build.gradle
+++ b/build.gradle
@@ -56,16 +56,34 @@ subprojects {
         configurations = []
         classifier = null
 
-        // Keep docker-java's package inside the final jar
-        relocate("com.github.dockerjava", "com.github.dockerjava")
-    }
+        doFirst {
+            // See https://github.com/johnrengelman/shadow/blob/5.0.0/src/main/groovy/com/github/jengelman/gradle/plugins/shadow/tasks/ConfigureShadowRelocation.groovy
+            Set<String> packages = []
+            // Always read from core's configurations
+            for (configuration in tasks.getByPath(":testcontainers:shadowJar").configurations) {
+                for (jar in configuration.files) {
+                    def jf = new java.util.jar.JarFile(jar)
+                    for (entry in jf.entries()) {
+                        def name = entry.name
+                        if (name.endsWith(".class")) {
+                            packages.add(name.substring(0, name.lastIndexOf('/')))
+                        }
+                    }
+                    jf.close()
+                }
+            }
+            for (pkg in packages) {
+                pkg = pkg.replaceAll('/', '.')
 
-    task relocateShadowJar(type: com.github.jengelman.gradle.plugins.shadow.tasks.ConfigureShadowRelocation) {
-        target = tasks.shadowJar
-        prefix = "org.testcontainers.shaded"
-    }
+                if (pkg.startsWith("com.github.dockerjava.")) {
+                    // Keep docker-java's package inside the final jar
+                    continue;
+                }
 
-    tasks.shadowJar.dependsOn tasks.relocateShadowJar
+                tasks.shadowJar.relocate(pkg, "org.testcontainers.shaded.${pkg}")
+            }
+        }
+    }
 
     publishing {
         publications {

